docs/source/kinds_of_types.rst

```diff
@@ -701,3 +701,207 @@ For more details, see :ref:`type-variable-value-restriction`.
 
    How ``bytes``, ``str``, and ``unicode`` are handled between Python 2 and
    Python 3 may change in future versions of mypy.
+
+.. _generators:
+
+Generators
+**********
+
+A generator can be annotated by the generic type
+``Generator[YieldType, SendType, ReturnType]``. For example:
+
+.. code-block:: python
+
+   def echo_round() -> Generator[int, float, str]:
+       sent = yield 0
+       while sent >= 0:
+           sent = yield round(sent)
+       return 'Done'
+
+Note that unlike many other generics in the typing module, the ``SendType`` of
+``Generator`` behaves contravariantly, not covariantly or invariantly.
+
+If your generator will only yield values, set the ``SendType`` and
+``ReturnType`` to None:
+
+.. code-block:: python
+
+   def infinite_stream(start: int) -> Generator[int, None, None]:
+       while True:
+           yield start
+           start += 1
+
+Alternatively, you may annotate your generator as having a return type of
+``Iterator[YieldType]``:
+
+.. code-block:: python
+
+   def infinite_stream(start: int) -> Iterator[int]:
```

Comment:
This example is a little perverse. Maybe something trivial e.g. `yield 0; yield 1` would be sufficient?